web/src/app/administration/components/classification-list/classification-list.component.spec.ts

```diff
@@ -0,0 +1,198 @@
+    getCustomisation: jest.fn().mockReturnValue(of())
+  })
+);
+const domainServiceSpy = jest.fn().mockImplementation(
+  (): Partial<DomainService> => ({
+    getSelectedDomainValue: jest.fn().mockReturnValue(of()),
+    getSelectedDomain: jest.fn().mockReturnValue(of())
+  })
+);
+const getImportingFinishedFn = jest.fn().mockReturnValue(of(true));
+const importExportServiceSpy = jest.fn().mockImplementation(
+  (): Partial<ImportExportService> => ({
+    getImportingFinished: getImportingFinishedFn
+  })
+);
+
+describe('ClassificationListComponent', () => {
+  let fixture: ComponentFixture<ClassificationListComponent>;
+  let debugElement: DebugElement;
+  let component: ClassificationListComponent;
+  let store: Store;
+  let actions$: Observable<any>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [NgxsModule.forRoot([ClassificationState, EngineConfigurationState])],
+      declarations: [
+        ClassificationListComponent,
+        ClassificationTypesSelectorStub,
+        SpinnerStub,
+        TreeStub,
+        SvgIconStub,
+        ImportExportStub,
+        InputStub
+      ],
+      providers: [
+        { provide: ClassificationsService, useClass: classificationServiceSpy },
+        { provide: ClassificationCategoriesService, useClass: classificationCategoriesServiceSpy },
+        { provide: DomainService, useClass: domainServiceSpy },
+        { provide: ImportExportService, useClass: importExportServiceSpy }
+      ]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(ClassificationListComponent);
+    debugElement = fixture.debugElement;
+    component = fixture.debugElement.componentInstance;
+    store = TestBed.inject(Store);
+    actions$ = TestBed.inject(Actions);
+    store.reset({
+      ...store.snapshot(),
+      classification: classificationStateMock,
+      engineConfiguration: engineConfigurationMock
+    });
+    fixture.detectChanges();
+  }));
+
+  it('should create component', () => {
+    expect(component).toBeTruthy();
+  });
+
+  /* HTML: ACTION TOOLBAR */
+  it('should call CreateClassification when add-classification button is clicked', async () => {
+    const button = debugElement.nativeElement.querySelector('.add-classification-button');
+    expect(button).toBeTruthy();
+    let actionDispatched = false;
+    actions$.pipe(ofActionDispatched(CreateClassification)).subscribe(() => (actionDispatched = true));
+    button.click();
+    expect(actionDispatched).toBe(true);
+  });
+
+  it('should display import-export component', () => {
+    expect(debugElement.nativeElement.querySelector('taskana-administration-import-export')).toBeTruthy();
+  });
+
+  it('should display classification-types-selector component', () => {
+    const typesSelectorComponent = debugElement.nativeElement.querySelector(
+      'taskana-administration-classification-types-selector'
+    );
+    expect(typesSelectorComponent).toBeTruthy();
+  });
```

Comment:
hier das element in einem Variable gespeichert

Reply:
This selector is so long that it wouldn't fit into one line. The other selector is shorter.

Reply:
ok